public/app/features/explore/TraceView/TraceView.tsx

```diff
@@ -25,6 +25,9 @@ type Props = {
 };
 
 export function TraceView(props: Props) {
+  if (!props.trace) {
```

Comment:
I think this is not a proper fix. We already has this check in line 84.
What would make sense is to change the trace type in line 23 to be optional and follow the typescript error path to https://github.com/grafana/grafana/blob/davkal/fix-missing-trace/packages/jaeger-ui-components/src/model/transform-trace-data.tsx#L74 and change it to something like this.

```
export default function transformTraceData(data?: TraceData & { spans: TraceSpanData[] }): Trace | null {
  if (!data?.traceID) {
    return null;
  }
  const traceID = data.traceID.toLowerCase();
```

Reply:
I think it actually makes sense to bail out as quickly as possible. If we do not have a trace there is nothing to render and it does not matter where it actually fails.

Reply:
But we could probably remove the later check because at that point we should be sure that it's not undefinend

Reply:
Or not really the transform do some more check, not sure if it could happen but seems like in addition if the trace does not have traceID then we also don't render anything so the second check should still be kept.

Reply:
But maybe both checks can be lifted up so we check all the requirements as the first step instead of running any hooks when we know we won't render anything anyway.